scenarios/bigquery.py

```diff
@@ -38,83 +40,96 @@ def check(*cmd, **keyargs):
+        raise ValueError(name)
 
 def do_query(query, out_filename, project):
     """Executes a bigquery query, outputting the results to a file."""
     with open(out_filename, 'w') as out_file:
-        check(['bq', 'query', '--format=prettyjson', '--project_id='+project], stdin=query,
-              stdout=out_file)
+        cmd = [
+            'bq', 'query', '-n100000', '--format=prettyjson', '--project_id=%s' % project]
+        check(cmd, stdin=query, stdout=out_file)
 
 def do_jq(jqfilter, data_filename, out_filename, jq_bin='jq'):
-    """Executes a jq command on a data file and outputs the results to a file."""
+    """Executes jq on a data file and outputs the results to a file."""
     with open(out_filename, 'w') as out_file:
         check([jq_bin, jqfilter, data_filename], stdout=out_file)
 
-def run_metric(project, bucket_path, metric, query, jqfilter):
-    """Executes a query, filters the results, and uploads filtered and complete results to GCS."""
-    fulldata_filename = metric + time.strftime('-%Y-%m-%d.json')
-    latest_filename = metric + '-latest.json'
-
-    do_query(query, fulldata_filename, project)
-    do_jq(jqfilter, fulldata_filename, latest_filename)
-    check(['gsutil', '-h', 'Cache-Control:private, max-age=0, no-transform', 'cp',
-           fulldata_filename, latest_filename, bucket_path + metric + '/'])
-
-def main(configs, project, bucket_path):
+def run_metric(project, prefix, metric, query, jqfilter):
+    """Runs query and filters results, uploading data to GCS."""
+    stamp = time.strftime('%Y-%m-%d')
+    raw = 'raw-%s.json' % stamp
+    filtered = 'daily-%s.json' % stamp
+    latest = '%s-latest.json' % metric
+
+    do_query(query, raw, project)
+    do_jq(jqfilter, raw, filtered)
+    def copy(src, dst):
+        """Use gsutil to copy src to test with minimal caching."""
+        check([
+            'gsutil', '-h', 'Cache-Control:max-age=60', 'cp', src, dst])
+
+    copy(raw, os.path.join(prefix, metric, raw))
+    copy(filtered, os.path.join(prefix, metric, filtered))
+    copy(filtered, os.path.join(prefix, latest))
+
+def main(configs, project, bucket_path, nonsense=True):
     """Loads metric config files and runs each metric."""
     if not project:
-        raise ValueError('project-id cannot be an empty string.')
+        raise ValueError('project', project)
     if not bucket_path:
-        raise ValueError('bucket cannot be an empty string.')
-    if bucket_path[-1] != '/':
-        bucket_path += '/'
+        raise ValueError('bucket_path', bucket_path)
 
     # the 'bq show' command is called as a hack to dodge the config prompts that bq presents
     # the first time it is run. A newline is passed to stdin to skip the prompt for default project
     # when the service account in use has access to multiple projects.
-    check(['bq', 'show'], stdin='\n')
-
-    if not configs:
-        configs = [open(path) for path in glob.glob("test-infra/metrics/*.yaml")]
+    if nonsense:
```

Comment:
maybe replace with `if not os.path.exists(os.path.expanduser("~/.bigqueryrc")):`, which also helps hint at *why* we're executing this weird bit of code.